File: spring-webflux-plain-java-example/src/main/java/de/cfranzen/examples/webflux/groups/WebClientConfiguration.java
package de.cfranzen.examples.webflux.groups;

import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.web.reactive.function.client.WebClient;

@Configuration
public class WebClientConfiguration {

    @Bean
    public WebClient webClient(WebClient.Builder builder, ClientProperties properties) {
        return builder
                .baseUrl(properties.url())
                .build();
    }

}

File: spring-webflux-kotlin-example/src/main/kotlin/de/cfranzen/examples/webflux/kotlin/groups/WebClientConfiguration.kt
package de.cfranzen.examples.webflux.kotlin.groups

import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.web.reactive.function.client.WebClient

@Configuration
class WebClientConfiguration {

    @Bean
    fun webClient(builder: WebClient.Builder, properties: ClientProperties) =
        builder
            .baseUrl(properties.url)
            .build()
}
